src/server/http/version_check.js

```diff
@@ -0,0 +1,27 @@
+import { badRequest } from 'boom';
+
+export default function (kbnServer, server, config) {
+  const versionHeader = 'kbn-version';
+  const actualVersion = config.get('pkg.version');
+  const versionCheckDisabled = config.get('server.disableVersionCheck');
+  const xsrfProtectionDisabled = config.get('server.xsrf.disableProtection');
+  // this accepts the xsrf config strictly for legacy reasons
+  const disabled = versionCheckDisabled || xsrfProtectionDisabled;
```

Comment:
Do we really need this? I realize previously we had conflated version checking and xsrf, but we always exposed it as an xsrf check. The version check is new, and users could legitimately want to skip xsfr check, but not the version check. This prevents them from choosing one or the other.